web/src/settings_users.js

```diff
@@ -176,18 +180,24 @@ function create_role_filter_dropdown($events_container, section) {
     }).setup();
 }
 
-function populate_users() {
+async function populate_users() {
     const active_user_ids = people.get_realm_active_human_user_ids();
     const deactivated_user_ids = people.get_non_active_human_ids();
 
     if (active_user_ids.length === 0 && deactivated_user_ids.length === 0) {
         failed_listing_users();
+        return;
     }
 
-    section.active.create_table(active_user_ids);
-    section.deactivated.create_table(deactivated_user_ids);
-    create_role_filter_dropdown($("#admin-user-list"), section.active);
-    create_role_filter_dropdown($("#admin-deactivated-users-list"), section.deactivated);
+    try {
+        await fetch_presence_user_setting();
+        section.active.create_table(active_user_ids);
+        section.deactivated.create_table(deactivated_user_ids);
+        create_role_filter_dropdown($("#admin-user-list"), section.active);
+        create_role_filter_dropdown($("#admin-deactivated-users-list"), section.deactivated);
```

Comment:
We don't need to use an async function here. Let's just pass these additional steps post fetch in the success callback.